tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -635,4 +635,48 @@ function() {
 		$tag = wp_get_attachment_image( $attachment_id, 'medium', false, array( 'class' => "wp-image-{$attachment_id}" ) );
 		$this->assertNotSame( $tag, webp_uploads_img_tag_update_mime_type( $tag, 'the_content', $attachment_id ) );
 	}
+
+	/**
+	 * Tests that the fallback script is added when a post with updated images is rendered.
+	 *
+	 * @test
+	 */
+	public function it_should_add_fallback_script_if_content_has_updated_images() {
+		$attachment_id = $this->factory->attachment->create_upload_object(
+			TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg'
+		);
+
+		apply_filters(
+			'the_content',
+			sprintf(
+				'<p>before image</p>%s<p>after image</p>',
+				wp_get_attachment_image( $attachment_id, 'medium', false, array( 'class' => "wp-image-{$attachment_id}" ) )
+			)
+		);
+
+		$this->assertTrue( has_action( 'wp_footer', 'webp_uploads_wepb_fallback' ) === 10 );
+
+		ob_start();
+		wp_footer();
+		$footer = ob_get_clean();
+
+		$this->assertStringContainsString( 'data:image/webp;base64,UklGR', $footer );
+	}
+
+	/**
+	 * Tests that the fallback script is not added when a post with no updated images is rendered.
+	 *
+	 * @test
+	 */
+	public function it_should_not_add_fallback_script_if_content_has_no_updated_images() {
+		apply_filters( 'the_content', '<p>no image</p>' );
+
+		$this->assertFalse( has_action( 'wp_footer', 'webp_uploads_wepb_fallback' ) );
+
+		ob_start();
+		wp_footer();
+		$footer = ob_get_clean();
```

Comment:
```suggestion
		$footer = get_echo( 'wp_footer' );
```

The WordPress test suite includes a helper for doing this, see [source code](https://github.com/WordPress/wordpress-develop/blob/80380cd3742c623dbe8d7a7c8384e7ceb74ccf4b/tests/phpunit/includes/utils.php#L351-L355).

🔢 Applies to other tests in this file too.